client/asset/eth/nodeclient_harness_test.go

```diff
@@ -158,129 +214,263 @@ func getContractAddrFromFile(fileName string) (common.Address, error) {
-		// testing harness.
-		err := os.MkdirAll(simnetWalletDir, 0755)
-		if err != nil {
-			return 1, fmt.Errorf("error creating simnet wallet dir dir: %v", err)
-		}
-		err = os.MkdirAll(participantWalletDir, 0755)
-		if err != nil {
-			return 1, fmt.Errorf("error creating participant wallet dir: %v", err)
-		}
-		ethSwapContractAddr, err = getContractAddrFromFile(ethSwapContractAddrFile)
-		if err != nil {
-			return 1, err
-		}
-		dexeth.ContractAddresses[0][dex.Simnet] = ethSwapContractAddr
-		fmt.Printf("ETH swap contract address is %v\n", ethSwapContractAddr)
-		tokenSwapContractAddr, err = getContractAddrFromFile(tokenSwapContractAddrFile)
-		if err != nil {
-			return 1, err
-		}
-		dexerc20.ContractAddresses[0][dex.Simnet] = tokenSwapContractAddr
-		fmt.Printf("Token swap contract addr is %v\n", tokenSwapContractAddr)
-		testTokenContractAddr, err = getContractAddrFromFile(testTokenContractAddrFile)
-		if err != nil {
-			return 1, err
-		}
-		fmt.Printf("Test token contract addr is %v\n", testTokenContractAddr)
-		err = setupWallet(simnetWalletDir, simnetWalletSeed, "localhost:30355")
-		if err != nil {
-			return 1, err
-		}
-		err = setupWallet(participantWalletDir, participantWalletSeed, "localhost:30356")
-		if err != nil {
-			return 1, err
-		}
-		ethClient, err = newNodeClient(getWalletDir(simnetWalletDir, dex.Simnet), dex.Simnet, tLogger.SubLogger("initiator"))
-		if err != nil {
-			return 1, fmt.Errorf("newNodeClient initiator error: %v", err)
-		}
-		if err := ethClient.connect(ctx); err != nil {
-			return 1, fmt.Errorf("connect error: %v\n", err)
-		}
-		defer ethClient.shutdown()
+func runSimnet(m *testing.M) (int, error) {
+	// Create dir if none yet exists. This persists for the life of the
+	// testing harness.
+	err := os.MkdirAll(simnetWalletDir, 0755)
+	if err != nil {
+		return 1, fmt.Errorf("error creating simnet wallet dir dir: %v", err)
+	}
+	err = os.MkdirAll(participantWalletDir, 0755)
+	if err != nil {
+		return 1, fmt.Errorf("error creating participant wallet dir: %v", err)
+	}
+	ethSwapContractAddr, err = getContractAddrFromFile(ethSwapContractAddrFile)
+	if err != nil {
+		return 1, err
+	}
+	dexeth.ContractAddresses[0][dex.Simnet] = ethSwapContractAddr
+	fmt.Printf("ETH swap contract address is %v\n", ethSwapContractAddr)
+	tokenSwapContractAddr, err = getContractAddrFromFile(tokenSwapContractAddrFile)
+	if err != nil {
+		return 1, err
+	}
+	dexerc20.ContractAddresses[0][dex.Simnet] = tokenSwapContractAddr
+	fmt.Printf("Token swap contract addr is %v\n", tokenSwapContractAddr)
+	testTokenContractAddr, err = getContractAddrFromFile(testTokenContractAddrFile)
+	if err != nil {
+		return 1, err
+	}
+	fmt.Printf("Test token contract addr is %v\n", testTokenContractAddr)
+	err = setupWallet(simnetWalletDir, simnetWalletSeed, "localhost:30355", dex.Simnet)
+	if err != nil {
+		return 1, err
+	}
+	err = setupWallet(participantWalletDir, participantWalletSeed, "localhost:30356", dex.Simnet)
```

Comment:
I think it's fine so everyone can test. Would need to be a fairly technical person to jack the testnet coins, and then I don't think they are worth real money... so no incentive?

I can drain the funds and do something else if you like. Some of the tests don't need funds and would work anyway. For instance `TestBasicRetrieval`

Reply:
I don't care about the funds that are in there presently.  But also what happens when two devs are using the same wallet at the same time?  I have a hard time seeing this work well on a public testnet.

I'm trying to understand if there's a way to have this work on testnet but with a seed that can be specified locally, either by editing the code or by an `-ldflags` switch.

Reply:
What if if the testnet seed/key could be specified locally, and there were a special test like `TestCreateTestnetWallet` that you could run to make the wallet and native light node, and it would spit out an address, then you could fund it, wait for confs, then run the other testnet tests now that there is funding there.

Reply:
Alternative the testnet tests could look for some text file in an expected location for the seed?

Reply:
> But also what happens when two devs are using the same wallet at the same time?

Yeah, that would cause unexpected results.

Reply:
Ok. Will add test that tests for funds and fails if there are not enough to run tests that need funds. It will also spit out the addresses that need funds.

And can the testnet seed just be a separate const? I already have the `testnet` switch as a constant... and the peer. So, following that pattern.

Reply:
https://github.com/decred/dcrdex/compare/e73f115596fee648b884e6bd6d1a955f83718482..271de4a504d8f0e444d37161b1b1cce0d0e2a9f8